### ISSUE TITLE
POI Gas Station Widget

### ISSUE THREAD

**Issue by Sonwon1** [osmandapp/OsmAnd#18007]
### Description

POI Gas Station Widget that would show the distance to the next 1 or 2 gas stations within 1-2 kilometers of the route.

### Describe the solution you'd like

POI Gas Station Widget that would show the distance to the next 1 or 2 gas stations within 1-2 kilometers of the route. This widget would go on the left or right panel. Sometimes a gas station is no longer there which is why 2 would be useful.

Would it be to slow to be useful?

### Describe alternatives you've considered

Show POI along route does not show the distance to the next gas station without digging into the menus.

**Comment by mikehgentry** [comment]
It would perhaps be more useful to generalise this (bike tourers for example care more about pubs or toilets than they do about petrol stations...)

So maybe make a widget that shows distance to next visible POI along the route (with a configurable tolerance for deviation). There could be a second widget showing distance to second next, or perhaps an alternative mode for the same widget when you tap it, to help decide how urgent a stop is. You could then handle which POIs you're interested in through quick actions.

Or if that seems too convoluted, maybe just make which POI it points to configurable in the options.

I think it'd also be useful to have an option to show the distance to the nearest (/second nearest) POI irrespective of direction. 

#17563 is related

**Comment by Sonwon1** [comment]
Also being able to tap on widget and then offer to route you there.

**Comment by DmitryAlexei** [comment]
The issue might be resolved by adding Quick Action with a custom POI search
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/7671b33b-4d2d-48cb-a42a-9586a6893958"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by smajser** [comment]
> The issue might be resolved by adding Quick Action with a custom POI search
> <img alt="" width="300" height="auto" src="https://github.com/user-attachments/assets/7671b33b-4d2d-48cb-a42a-9586a6893958">

I have created Quick Action for Filling stations, but how can I select, for example, only OMV gas stations to be shown with quick action?

**Comment by DmitryAlexei** [comment]
You can create custom POI search for Filling stations OMV (for more information see https://download.osmand.net/docs/user/search/search-poi#custom-poi-search). After that, create a single action Quick Action that shows / hides these POIs

**Comment by Sonwon1** [comment]
This one is over two years old?

**Comment by Sonwon1** [comment]
> You can create custom POI search for Filling stations OMV (for more information see https://download.osmand.net/docs/user/search/search-poi#custom-poi-search). After that, create a single action Quick Action that shows / hides these POIs

Yes you can however then you have to stop driving while you play around with finding the next POI you want. I am looking to increase the user friendliness.

**Comment by C6competition** [comment]
I would as well appreciate, if the POI Gas Station Widget becomes realised!
